storm-core/src/jvm/org/apache/storm/task/OutputCollector.java

```diff
@@ -218,6 +218,16 @@ public void fail(Tuple input) {
         _delegate.fail(input);
     }
 
+    /**
+    * Resets the message timeout for any tuple trees to which the given tuple belongs.
+    * The timeout is reset to Config.TOPOLOGY_MESSAGE_TIMEOUT_SECS.
+    * @param input the tuple to reset timeout for
+    */
```

Comment:
Can we add to the javadoc that this should be used sparingly?  This is not a cheap operation, so having someone do it for each tuple they receive is very expensive.